### PR TITLE
Updated placeOrder mutation for negotiable quote usage

### PR DESCRIPTION
**Pull request by amolina-magento**
## Problem

The existing placeOrder mutation works for Cart, but does not account for Negotiable Quote usage.

## Solution

Override the mutation to support Negotiable Quotes.

## Requested Reviewers

@cpartica

### CODE REVIEW

**Comment by nmarquez2** [comment]
@amolina-magento I don't think we should override the placeOrder mutation itself, we could extend the existing PlaceOrderInput type to make cart_id optional and add a quote_uid optional parameter so the mutation parameters can be consistent with other Negotiable Quote Mutations. 

After that we can do a simple plugin on the placeOrder resolver to perform Negotiable Quote validations and ensure either cart_id or quote_uid is provided and modify the input accordingly to allow the original method to run normally.

**Comment by nmarquez2** [comment]
After internal discussions we concluded that we need to have a precise separate mutation to place an order from a negotiable quote to retain the clear distinction between quote types available actions. The PR was adjusted to reflect that by adding the placeNegotiableQuoteOrder mutation with a PlaceNegotiableQuoteOrderInput type that keeps the input consistent with other NQ mutations.

File: design-documents/graph-ql/coverage/b2b/negotiableQuotes.graphqls [hunk 1]
```diff
@@ -71,6 +71,10 @@ type Mutation {
         input: SetNegotiableQuotePaymentMethodInput!
     ): SetNegotiableQuotePaymentMethodOutput @doc(description: "Set the payment method on the negotiable quote")
 
+    placeNegotiableQuoteOrder(
+        input: PlaceNegotiableQuoteOrderInput
+    ): PlaceOrderOutput @doc(description: "Place an order using the negotiable quote")
```
Comment (pdohogne-magento) [inline design-documents/graph-ql/coverage/b2b/negotiableQuotes.graphqls:76]: This should have a NQ-specific output class as well.

Comment (nmarquez2) [reply design-documents/graph-ql/coverage/b2b/negotiableQuotes.graphqls]: Schema has been updated

**Review by keharper — APPROVED** [review]

**Review by prabhuram93 — APPROVED** [review]

**Review by pdohogne-magento — APPROVED** [review]

**Pull request merged**